presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -216,7 +235,6 @@ private static PlanNode planTableScan(TableScanNode node, Expression predicate,
                 simplifiedConstraint.intersect(layout.getLayout().getPredicate()),
```

Comment:
Layout predicate might still be quite complex for selected partitions, e.g: OR of all dates from last few years (few hundreds/thousands entries). Consider simplifying predicate when it's too complex (e.g: more than 50 expression nodes).